### PR TITLE
Test/mds 1451

### PR DESCRIPTION
**Pull request by thegentlemanphysicist**
This is the first part of ticket 1451.  It creates tests that allow a user to upload and download a TSF test doc.  I've done a few things that are outside the normal scope of functional tests (downloading a doc is  not usually covered by browser automation) any feed back on the code structure is apreciated.
Current pipeline now runs headless firefox tests

### CODE REVIEW

File: functional-tests/src/test/groovy/spec/Tailings.groovy [hunk 1]
```diff
@@ -38,5 +50,46 @@ class Tailings extends GebReportingSpec {
...
+    def "Scenario: User is able to download a TSF Report"() {
+        when: "User navigates to the TSF tab and clicks the download icon"
+        tailingsTab.tabSelect.click()
+
+        and: "User opens a file in the folder specified in GebConfig"
+        tailingsTab.downloadLink[0].click()
+
+        def file = new File(Const.DOWNLOAD_PATH+'/'+Const.TEST_FILE_NAME)
+        //allow time for the file to be created in the DOWNLOAD_PATH
+        //throw an error if it takes more than 20sec
```
Comment (NWCalvank) [inline functional-tests/src/test/groovy/spec/Tailings.groovy]: Do we have any idea of how long this **can** take and then still be successful? Our other timeouts in the functional tests are 30 seconds, if I remember correctly.

File: functional-tests/src/test/groovy/spec/Tailings.groovy [hunk 2]
```diff
@@ -38,5 +50,46 @@ class Tailings extends GebReportingSpec {
...
+        def uploadedFile = dir.newFile(Const.TEST_FILE_NAME) << Const.TEST_FILE_CONTENT
+        tailingsTab.addTailingsDocButtons[0].click()
+        tailingsTab.uploadField = uploadedFile.absolutePath
+
+        then: "The doc upload complete message is shown"
+        assert tailingsTab.uploadCompleteMessage != null
+
+    }
+
+    //ToDo: THIS TEST ONLY WORKS IN CHROME AND CHROME HEADLESS.  WORK IN GebConfig to fix other browsers
```
Comment (NWCalvank) [inline functional-tests/src/test/groovy/spec/Tailings.groovy]: TODO label on comments should be all-caps

File: functional-tests/src/test/tempStorage/test (1).odt [hunk 3]
```diff
@@ -0,0 +1 @@
+The quality of mercy is not strained.
```
Comment (NWCalvank) [inline functional-tests/src/test/tempStorage/test (1).odt]: I tip my hat to this

File: functional-tests/src/test/groovy/spec/Tailings.groovy (hunk 1 above)
Comment (thegentlemanphysicist) [reply functional-tests/src/test/groovy/spec/Tailings.groovy]: I gave it a 20 second loop somewhat arbitrarily.  If it takes longer, to download than that, something has gone wrong.  I wanted it to be less than 30 seconds so there would be clear error message about what went wrong.

File: functional-tests/src/test/tempStorage/test (1).odt (hunk 3 above)
Comment (thegentlemanphysicist) [reply functional-tests/src/test/tempStorage/test (1).odt]: It droppeth as the gentle rain from heaven Upon the place beneath.

File: functional-tests/src/test/groovy/spec/Tailings.groovy (hunk 2 above)
Comment (thegentlemanphysicist) [reply functional-tests/src/test/groovy/spec/Tailings.groovy]: Changed

**Pull request merged**